src/halmos/sevm.py

```diff
@@ -2017,6 +2025,85 @@ def resolve_address_alias(
         ex.alias[target] = addr
         return addr
 
+    def refine_calldata(
+        self,
+        ex: Exec,
+        opcode: int,
+        to_alias: Address,
+        stack: list[tuple[Exec, int]],
+        step_id: int,
+    ):
+        arg_pos = 4 if opcode in [EVM.CALL, EVM.CALLCODE] else 3
+        arg_loc = ex.mloc(ex.st.peek(arg_pos))
+        arg_size = ex.int_of(ex.st.peek(arg_pos + 1), "symbolic CALL input data size")
+        if not arg_size >= 0:
+            raise ValueError(arg_size)
+        arg = ex.st.memory.slice(arg_loc, arg_loc + arg_size)
+
+        if to_alias is None:
+            return arg
+
+        if len(arg) < 4:
+            return arg
+
+        selector = arg[0:4].unwrap()
+
+        # TODO: check if calldata can be decoded. even if selector is concrete, remaining calldata may not be decodable.
+        if (
+            isinstance(selector, bytes)
+            or is_bv_value(selector)
+            or str(selector).startswith("fallback_selector_")
+        ):
+            return arg
```

Comment:
another issue: even if selector is symbolic, calldata may be decodable, in which case, refining calldata is not needed, and may lead to unexpected behaviors.